qiskit/algorithms/quantum_time_evolution/variational/calculators/evolution_grad_calculator.py

```diff
@@ -0,0 +1,88 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Calculates and binds evolution gradients for Variational Quantum Time Evolution."""
+
+from typing import Union, Optional, List, Dict, Callable
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector, ParameterExpression, Parameter
+from qiskit.opflow import StateFn, Gradient, CircuitGradient, OperatorBase, Z, CircuitSampler
+from qiskit.opflow.gradients.circuit_gradients import LinComb
+
+
+def calculate(
+    observable: OperatorBase,
+    ansatz: Union[StateFn, QuantumCircuit],
+    parameters: Optional[Union[ParameterVector, ParameterExpression, List[ParameterExpression]]],
+    grad_method: Union[str, CircuitGradient],
+    basis: OperatorBase = Z,
+) -> OperatorBase:
+    """
+    Calculates a parametrized evolution gradient object.
+    Args:
+        observable: Observable for which an evolution gradient should be calculated,
+                    e.g., a Hamiltonian of a system.
+        ansatz: Quantum state to be evolved.
+        parameters: Parameters with respect to which gradients should be computed.
+        grad_method: The method used to compute the state gradient. Can be either
+                    ``'param_shift'`` or ``'lin_comb'`` or ``'fin_diff'``.
+        basis: Basis with respect to which evolution gradient is calculated. In case of a default
+                Z basis a real part of an evolution gradient is returned. In case of a Y basis,
+                an imaginary part of an evolution gradient is returned.
+    Returns:
+        Parametrized evolution gradient as an OperatorBase.
+    """
+    operator = ~StateFn(observable) @ StateFn(ansatz)
+    if grad_method == "lin_comb":
+        return LinComb().convert(operator, parameters, aux_meas_op=basis)
+    return Gradient(grad_method).convert(operator, parameters)
+
+
+def eval_grad_result(
```

Comment:
Thanks, it might make sense to do what you suggest. It might be the case that this method is used more often in error bounds that were not included yet in the implementation. I will need to check it.